src/llv8.cc

```diff
@@ -220,10 +271,9 @@ std::string LLV8::LoadTwoByteString(int64_t addr, int64_t length, Error& err) {
     return std::string();
   }
 
-  for (int64_t i = 0; i < length; i++) buf[i] = buf[i * 2];
-  buf[length] = '\0';
+  char16_t* u16str = (char16_t*) buf;
```

Comment:
This isn't safe, unfortunately, because it aliases and `buf` need not be properly aligned (although it probably is since it was new'd.)

Wouldn't calling `new char16_t[length]` work?  Since you're here, maybe switch to `std::vector<char16_t>`.

Aside: `ConvertUTF16toUTF8()` uses `llvm::UTF16` (typedef for `unsigned short`), not `char16_t`.

Reply:
thanks for reviewing, I'll fix this